Update node

thomasleese opened the pull request:
Update the version of node available on most of the machines, but leave the performance platform on the older version as they are still dependant on it.

Closes https://github.com/alphagov/govuk-puppet/issues/5942

[Trello Card](https://trello.com/c/YPhTen16/1041-2-automate-the-building-of-documentation) [Trello Card](https://trello.com/c/M8uRSjYq/113-spike-npm-integration-into-static-2)

thomasleese commented:
Tested on integration, seems to be working:

```bash
❯ ssh jenkins-1.integration
Last login: Fri Aug 18 13:12:21 2017 from jumpbox-1.management.integration.publishing.service.gov.uk
thomasleese@integration-jenkins-1:~$ node --version
v6.11.2

❯ ssh performance-frontend-1.integration
Last login: Fri Aug 18 13:14:55 2017 from jumpbox-1.management.integration.publishing.service.gov.uk
thomasleese@integration-performance-frontend-1:~$ node --version
v0.10.37
```

cc @surminus @nickcolley

nickcolley commented:
I think the only place we current use Node.js on the main machines is in Sprockets JavaScript uglification, so we'd want to check the JavaScript output potentially for an application.

thomasleese commented:
Just checked the digests of the assets in integration and production, and they match.

thomasleese commented:
Plus performance platform still seems to be working fine in integration.

nickcolley commented:
Looks good to me 👍 

(Will await someone who knows how Puppet works :))

surminus commented on modules/nodejs/manifests/init.pp:
```diff
@@ -18,6 +18,8 @@
+  class { '::nodejs::repo': }
+
   package { 'nodejs':
     ensure  => $ensure,
```
You should add a dependency for the repo:
`require => Class['Nodejs::Repo'],`

surminus commented on modules/nodejs/manifests/repo.pp:
```diff
@@ -0,0 +1,20 @@
+    location     => "http://${apt_mirror_hostname}/nodejs",
+    release      => $::lsbdistcodename,
+    architecture => $::architecture,
+    key          => '',
```
The key should be the same as all the others. Check across the codebase for where we have used `apt::source` previously.

surminus commented:
Couple of comments but it looks good! I think we install statsd with npm, so please check that's OK (which I'm sure it is).

thomasleese commented:
@surminus Thanks, I've addressed your comments. Regarding statsd, I saw that it installed correctly, and the following command returns something good:

```
thomasleese@integration-jenkins-1:~$ sudo service statsd status
statsd start/running, process 785
```

Anything more I could check?

surminus commented:
We use statsd for counters like number of status codes appearing in nginx for example, so I guess just check that graphite for these metrics don't disappear.

thomasleese commented:
@surminus Just checked in graphite, and there does appear to be data for the last hour in there.
